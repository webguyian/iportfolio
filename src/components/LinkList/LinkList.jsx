import React from 'react';
import { array } from 'prop-types';

import Link from 'components/Link/Link';
import Text from 'components/Text/Text';

const LinkList = props => {
  const { links } = props;
  const baseClass = 'ui-link-list';
  const itemClass = `${baseClass}-item`;
  const linkClass = `${itemClass}-link`;

  return (
    <ul className={baseClass}>
      {links.map(link => (
        <li key={link.abbr} className={itemClass}>
          <Link className={linkClass} to={link.url} external>
            <div
              className={`${linkClass}-icon`}
              style={{ backgroundColor: link.color }}
            >
              <abbr title={link.name}>{link.abbr}</abbr>
            </div>
            <Text className={`${linkClass}-label`}>{link.name}</Text>
          </Link>
        </li>
      ))}
    </ul>
  );
};

LinkList.propTypes = {
  links: array.isRequired
};

export default LinkList;
