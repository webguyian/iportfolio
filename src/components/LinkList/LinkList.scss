.ui-link-list {
  @include unstyled-list;
  display: grid;
  grid-gap: $gutter-s 0;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr;
  margin-bottom: $gutter-l;

  &-item-link {
    @include focus-outline;
    display: block;
    padding: $gutter-s;
    text-align: center;

    &:hover,
    &:focus {
      .ui-link-list-item-link-icon {
        transform: scale(1.09);
      }
    }

    &-icon {
      @include circle(100px, $border-radius-m);
      @include flex-center;
      transition: transform 0.2s ease-in;

      > abbr {
        @include type-display($font-size-xl);
        text-decoration: none;
      }
    }

    &-label {
      @include type-body(12px);
      display: block;
      margin-top: $gutter-s;
    }
  }
}
