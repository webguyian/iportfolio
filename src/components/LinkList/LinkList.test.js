import React from 'react';
import { render } from '@testing-library/react';

import LinkList from './LinkList';

describe('<LinkList />', () => {
  const props = {
    links: [
      {
        abbr: 'MD',
        color: '#83d0f2',
        name: 'MDN',
        url: 'https://developer.mozilla.org/en-US/'
      },
      {
        abbr: 'SO',
        color: '#f48024',
        name: 'Stack Overflow',
        url: 'https://stackoverflow.com/'
      }
    ]
  };

  it('renders correctly', () => {
    const { asFragment } = render(<LinkList {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
