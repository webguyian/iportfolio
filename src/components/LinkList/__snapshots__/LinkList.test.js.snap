// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkList /> renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="ui-link-list"
  >
    <li
      class="ui-link-list-item"
    >
      <mock-link
        classname="ui-link-list-item-link"
        external="true"
        to="https://developer.mozilla.org/en-US/"
      >
        <div
          class="ui-link-list-item-link-icon"
          style="background-color: rgb(131, 208, 242);"
        >
          <abbr
            title="MDN"
          >
            MD
          </abbr>
        </div>
        <span
          class="ui-text ui-text--body ui-link-list-item-link-label"
        >
          MDN
        </span>
      </mock-link>
    </li>
    <li
      class="ui-link-list-item"
    >
      <mock-link
        classname="ui-link-list-item-link"
        external="true"
        to="https://stackoverflow.com/"
      >
        <div
          class="ui-link-list-item-link-icon"
          style="background-color: rgb(244, 128, 36);"
        >
          <abbr
            title="Stack Overflow"
          >
            SO
          </abbr>
        </div>
        <span
          class="ui-text ui-text--body ui-link-list-item-link-label"
        >
          Stack Overflow
        </span>
      </mock-link>
    </li>
  </ul>
</DocumentFragment>
`;
